src/internal/interpretor.js

```diff
@@ -40,14 +40,14 @@ export default function interpretor(rej, res){
   //It will tell the current action that the future rejected, and it will
   //settle the current tick with the action's answer to that.
   function rejected(x){
-    settle(action.rejected(x));
+    settle(action && action.rejected(x));
```

Comment:
Why this line should be reverted: This line is now buggy. If action would be falsy, we would be passing a falsy value into `settle` which will lead to more breakage. We *could* "fix" it like this: `action && settle(action.rejected(x))`, but that will cause the Future to hang when the action is missing.

More importantly there should be no code paths allowing for `action` to be anything else than an instance of `Action`. I suspect from your bug report that there actually *is* a code path like that hiding somewhere, which means we should find it and fix it, rather than implementing a null check.